src/main/kotlin/uk/gov/dluhc/printapi/service/PrintResponseBatchService.kt

```diff
@@ -0,0 +1,31 @@
+package uk.gov.dluhc.printapi.service
+
+import uk.gov.dluhc.printapi.database.entity.Status.PENDING_ASSIGNMENT_TO_BATCH
+import uk.gov.dluhc.printapi.database.entity.Status.RECEIVED_BY_PRINT_PROVIDER
+import uk.gov.dluhc.printapi.database.entity.Status.SENT_TO_PRINT_PROVIDER
+import uk.gov.dluhc.printapi.database.repository.PrintDetailsRepository
+import uk.gov.dluhc.printapi.printprovider.models.BatchResponse
+import uk.gov.dluhc.printapi.printprovider.models.BatchResponse.Status.SUCCESS
+
+class PrintResponseBatchService(
+    private val printDetailsRepository: PrintDetailsRepository,
+    private val idFactory: IdFactory
+) {
+    fun processBatchResponses(batchResponses: List<BatchResponse>) {
```

Comment:
Love it; this is exactly what I had in mind ❤️ (ie. not getting and updating individually 👍 )